### PR TITLE
external-idp: change idp server name to reference name

### PR DESCRIPTION
**Pull request by tscherf**
When you  run "ipa idp-show <idp reference>" the IdP reference is shown as "Identity Provider server name". This is confusing as we are pointing to the earlier created IdP reference rather than a server.

### CODE REVIEW

**Comment by abbra** [comment]
@tscherf thanks. There are few more places to change. Please update the doc in `ipaserver/plugins/idp.py` as that contains the same 'server name' part and there are few more places which reference those 'server name' in other parameters. Also permissions have 'server' in the name but I don't think we should change those.

Luckily, these parameters' labels aren't visible in API.txt so they aren't part of the API change.

```
diff --git a/ipaserver/plugins/idp.py b/ipaserver/plugins/idp.py
index 2a3cd9f28..9e5fc70a2 100644
--- a/ipaserver/plugins/idp.py
+++ b/ipaserver/plugins/idp.py
@@ -22,34 +22,34 @@ from itertools import chain
 logger = logging.getLogger(__name__)
 
 __doc__ = _("""
-External Identity Provider Servers
+External Identity Provider References
 """) + _("""
-Manage External Identity Provider Servers.
+Manage External Identity Provider References.
 """) + _("""
-IPA supports the use of an external Identity Provider for Oauth2.0 Device Flow
+IPA supports the use of an external Identity Provider for OAuth2.0 Device Flow
 authentication.
 """) + _("""
 EXAMPLES:
 """) + _("""
- Add a new external Identity Provider server:
+ Add a new external Identity Provider reference:
    ipa idp-add MyIdP --client-id jhkQty13 \
       --auth-uri https://oauth2.idp.com/auth \
       --token-uri https://oauth2.idp.com/token --secret
 """) + _("""
- Add a new external Identity Provider server using github predefined endpoints:
+ Add a new external Identity Provider reference using github predefined endpoints:
    ipa idp-add MyIdp --client-id jhkQty13 --provider github --secret
 """) + _("""
- Find all external Identity Provider servers whose entries include the string
+ Find all external Identity Provider references whose entries include the string
  "test.com":
    ipa idp-find test.com
 """) + _("""
- Examine the configuration of an external Identity Provider server:
+ Examine the configuration of an external Identity Provider reference:
    ipa idp-show MyIdP
 """) + _("""
  Change the secret:
    ipa idp-mod MyIdP --secret
 """) + _("""
- Delete an external Identity Provider server:
+ Delete an external Identity Provider reference:
    ipa idp-del MyIdP
 """)
 
@@ -80,8 +80,8 @@ class idp(LDAPObject):
     Identity Provider object.
     """
     container_dn = api.env.container_idp
-    object_name = _('Identity Provider server')
-    object_name_plural = _('Identity Provider servers')
+    object_name = _('Identity Provider reference')
+    object_name_plural = _('Identity Provider references')
     object_class = ['ipaidp']
     default_attributes = [
         'cn', 'ipaidpauthendpoint', 'ipaidpdevauthendpoint',
@@ -95,13 +95,13 @@ class idp(LDAPObject):
         'ipaidpkeysendpoint', 'ipaidpscope', 'ipaidpsub',
     ]
     allow_rename = True
-    label = _('Identity Provider servers')
-    label_singular = _('Identity Provider server')
+    label = _('Identity Provider references')
+    label_singular = _('Identity Provider reference')
 
     takes_params = (
         Str('cn',
             cli_name='name',
-            label=_('Identity Provider server name'),
+            label=_('Identity Provider reference name'),
             primary_key=True,
             ),
         Str('ipaidpauthendpoint?',
```

Finally, there are few output references in `doc/workshop/12-external-idp-support.rst` that would need to be updated with this change.

**Comment by tscherf** [comment]
@abbra - I hope I've catched all the places where the name has to be changed. Fixed some typos too.

**Comment by flo-renaud** [comment]
@tscherf 
1. You need to rebase on top of the master branch as it contains a fix for the WebUI_Unit_Tests failure
2. You need to run `python makeapi.in` and commit the changes if any (the problem was detected by lint)

**Comment by abbra** [comment]
@tscherf could you please rebase?

File: ipaserver/plugins/idp.py [hunk 1]
```diff
@@ -22,34 +22,34 @@
...
 EXAMPLES:
 """) + _("""
- Add a new external Identity Provider server:
+ Add a new external Identity Provider reference:
    ipa idp-add MyIdP --client-id jhkQty13 \
       --auth-uri https://oauth2.idp.com/auth \
       --token-uri https://oauth2.idp.com/token --secret
 """) + _("""
- Add a new external Identity Provider server using github predefined endpoints:
+ Add a new external Identity Provider reference using github predefined endpoints:
```
Comment (flo-renaud) [inline ipaserver/plugins/idp.py:39]: The line is too long, you can add a break between `predefined` and `endpoints`.

**Review by flo-renaud — COMMENTED** [review]
Hi @tscherf 

-  Maybe you are using an editor that adds/removes newline at end of file? That's one of the difference I can see when I manually run python makeapi.in in my repo, and could explain why lint is complaining.
- No need to make changes to the po files as they would be overwritten when we receive new translations.
- line 39 is too long in idp.py, please see my inline comment.

**Review by flo-renaud — COMMENTED** [review]
Hi @tscherf 
The xmlrpc tests also need to be adapted, see for instance https://github.com/freeipa/freeipa/blob/9ee16e8acc1a980c7d9b4663625c1b76e474625d/ipatests/test_xmlrpc/test_idp_plugin.py#L58 where the exception message has changed.

**Comment by tscherf** [comment]
The file 'doc/api/idp_show.md' file is generated by 'makeapi'  but it keeps using 'Identity Provider server'. Where is this coming from? @flo-renaud

**Comment by flo-renaud** [comment]
> The file 'doc/api/idp_show.md' file is generated by 'makeapi' but it keeps using 'Identity Provider server'. Where is this coming from? @flo-renaud

@tscherf 
it's generated from these lines:
https://github.com/tscherf/freeipa/blob/idp-reference-name/ipaserver/plugins/idp.py#L442-L443

**Comment by tscherf** [comment]
@flo-renaud are there any other changes required before you can merge?

**Comment by flo-renaud** [comment]
Hi @tscherf 
thanks for your patch, LGTM. I didn't find additional occurrences of the string "Identity Provider server" so I think the PR is ready to be merged.
@abbra we need backports down to ipa-4-9, right?

**Comment by abbra** [comment]
Yes, we need 4.9 and 4.10 backports as this is where IdP integration is available.

**Comment by flo-renaud** [comment]
master:

* 2aeb963fc9f3261e5f9539450cef96b8dbf84135 external-idp: change idp server name to reference name

**Pull request closed without merging**